From a850dc21263bd8a323b50cb041ef60a9dae556ac Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Fri, 25 Jun 2010 13:55:49 -0400
Subject: [PATCH] ENGR00124569 IPUv3:Change for support VDI in IPU lib

1) Support select VDI input channel buffer ioctrl.
2) Add VDI input channels' EOF interrupt.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 drivers/mxc/ipu3/ipu_device.c |   11 +++++++++++
 include/linux/ipu.h           |    4 ++++
 2 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/drivers/mxc/ipu3/ipu_device.c b/drivers/mxc/ipu3/ipu_device.c
index 27455fe..7f7d829 100644
--- a/drivers/mxc/ipu3/ipu_device.c
+++ b/drivers/mxc/ipu3/ipu_device.c
@@ -183,6 +183,17 @@ static int mxc_ipu_ioctl(struct inode *inode, struct file *file,
 
 		}
 		break;
+	case IPU_SELECT_MULTI_VDI_BUFFER:
+		{
+			uint32_t parm;
+			if (copy_from_user
+				(&parm, (uint32_t *) arg,
+				sizeof(uint32_t)))
+				return -EFAULT;
+
+			ret = ipu_select_multi_vdi_buffer(parm);
+		}
+		break;
 	case IPU_LINK_CHANNELS:
 		{
 			ipu_channel_link link;
diff --git a/include/linux/ipu.h b/include/linux/ipu.h
index 05f003c..fbd209f 100644
--- a/include/linux/ipu.h
+++ b/include/linux/ipu.h
@@ -609,6 +609,9 @@ enum ipu_irq_line {
 	IPU_IRQ_CSI1_OUT_EOF = 1,
 	IPU_IRQ_CSI2_OUT_EOF = 2,
 	IPU_IRQ_CSI3_OUT_EOF = 3,
+	IPU_IRQ_VDI_P_IN_EOF = 8,
+	IPU_IRQ_VDI_C_IN_EOF = 9,
+	IPU_IRQ_VDI_N_IN_EOF = 10,
 	IPU_IRQ_PP_IN_EOF = 11,
 	IPU_IRQ_PRP_IN_EOF = 12,
 	IPU_IRQ_PRP_GRAPH_IN_EOF = 14,
@@ -1267,6 +1270,7 @@ typedef struct _ipu_csc_update {
 #define IPU_CALC_STRIPES_SIZE	      _IOWR('I', 0x27, ipu_stripe_parm)
 #define IPU_UPDATE_BUF_OFFSET         _IOW('I', 0x28, ipu_buf_offset_parm)
 #define IPU_CSC_UPDATE                _IOW('I', 0x29, ipu_csc_update)
+#define IPU_SELECT_MULTI_VDI_BUFFER   _IOW('I', 0x2A, uint32_t)
 
 int ipu_calc_stripes_sizes(const unsigned int input_frame_width,
 				unsigned int output_frame_width,
-- 
1.5.4.4

